### ISSUE TITLE
Unable to update item name with 'full stop' character in string

### ISSUE THREAD

**Issue by bentura** [pnp/pnpjs#2756]
### What version of PnPjs library you are using

2.x (No longer supported)

### Minor Version Number

2.15.0

### Target environment

SharePoint Framework

### Additional environment details

SPFX custom WebPart

### Question/Request

Hi,

I have a function in a custom WebPart that is updating the Title and FileLeafRef of an item.  Users are able to click on a rename button, enter a new name into a text box, and then save.  This works fine unless the users want to include a 'full stop' ( .) character in the name.  If the item is a file, then the new name gets cut off at the . character.   

For example, a word document called Test1, renamed to Test1.NewName would still only save as Test1.  If the user is renaming a folder then this works fine, it is only when the item is a file.

Is this expected behavour?  I know that when you add a new file, the file name can have the . character in it.

I know I am using v2 but I imagine is would be the same in v3

```TypeScript

  let list = sp.web.lists.getByTitle(globalThis.libraryNameShort);

  const i = await list.items.getById(item.Id).update({
    Title: newName,
    FileLeafRef: newName,
  }); 
```

in the above code, newName is a string containing the . character in the middle.

Thanks

**Comment by juliemturner** [comment]
If i were you I'd take a look at the payload of the call and see if the name with the . in it is being sent to the api endpoint. If it is, then that's how the API behaves, if it isn't then it may be an issue with parsing, but I don't really think we were doing anything like that in v2.

**Comment by bentura** [comment]
Hi,

Thank you for your quick reply.  I've had a look at the payload (think I'm looking int he right place) and can see that it looks like the filename is being sent correctly.  Does this mean that my issue is on the SharePoint side? 



![image](https://github.com/pnp/pnpjs/assets/25690352/6581f50d-853b-4fca-9f9a-52cdb2d11de5)

**Comment by juliemturner** [comment]
Yes, that's what that means. I would inspect how SharePoint updates the file name in the UI to see what it does differently.

**Comment by patrick-rodgers** [comment]
This is going to sound dumb, but can you try some other extension than ".file"?

**Comment by bentura** [comment]
Not dumb at all.  I could have chosen a better example.  It doesn't make a difference what I put.

However, I've found someone with a similar issue when using Power Automate.  They were advised to include the actual file extension on the end of the name and this has worked for them.  I did what @juliemturner recommended and looked at the payload from the default SharePoint rename, and that passes the file extension  (so you would have dummy.123.pdf.

I'm assuming SharePoint is working backawards when reading the text sent and removing everything after the first full stop it comes accross. That would explain the behaviour.

Thank you for you assistance on this.
